### ISSUE TITLE
Dependencies not working in demo website

### ISSUE THREAD

**Issue by alexandru-cazacu** [ngx-formly/ngx-formly#3438]
Hi,

the demo page at https://formly.dev/docs/examples/advanced/json-schema does't work for `schema_dependencies`.

I would expect that by selecting "Do you have any pets?" a new field "How old is your pet?" should show as per schema definition, but nothing happens.

![image](https://user-images.githubusercontent.com/25430815/190686629-54f0041f-3cd0-454f-8433-b8658cfbb40c.png)

![image](https://user-images.githubusercontent.com/25430815/190686772-6e6c8fa4-b581-465c-a192-ee3df2bafcc8.png)

**Comment by vezzoo** [comment]
Hi, I work with Alexandru and I took a look at the issue and I think there's an error inside [FormlyJsonschema](https://github.com/ngx-formly/ngx-formly/blob/main/src/core/json-schema/src/formly-json-schema.service.ts) service when the "hide" expression is being created. I refer to lines [303](https://github.com/ngx-formly/ngx-formly/blob/main/src/core/json-schema/src/formly-json-schema.service.ts#L303) and [311](https://github.com/ngx-formly/ngx-formly/blob/main/src/core/json-schema/src/formly-json-schema.service.ts#L311), where a function with the following properties is defined:

- An argument “m” (I suppose that is an abbreviation of "model")
- In the implementation tries to retrieve the value of another field by accessing "m"

The problem is that the argument passed to the lambda is the field, not the model. So I think that the solution is to retrieve the other field value by accessing "field.model":

- line 303:
  ```patch
  - hide: (m: any) => !m || getConstValue(constSchema as JSONSchema7) !== m[property],
  + hide: (f) => !f.model || getConstValue(constSchema as JSONSchema7) !== f.model[property], 
  ```
- line 311:
  ```patch
  - hide: (m: any) => !m || isEmpty(m[property]),
  + hide: (f) => !f.model || isEmpty(f.model[property]),
  ```

Note that "m: any" has been replaced with "f" and "m" with "f.model".

Let me know if you want me to open a pull request.

**Comment by aitboudad** [comment]
>Let me know if you want me to open a pull request.

Yes,  please!

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.0.1](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.0.1) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
